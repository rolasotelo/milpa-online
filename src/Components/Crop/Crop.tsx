import React from "react";
import { flatten, map, pluck, without } from "underscore";
import useGameContext from "../../Hooks/useGameContext/useGameContext";
import { ModifierId } from "../../Realms/Pure/enums";
import { BoardSlot } from "../../Realms/Pure/types";

interface Props {
  boardSlot: Readonly<BoardSlot>;
  canInteract: boolean;
  isYourBoard: boolean;
}

const iconDictionary: Map<string, string> = new Map<string, string>();
iconDictionary.set(ModifierId.Huitlacoche, "🍄");
iconDictionary.set(ModifierId.Tuna, "🍓");
iconDictionary.set(ModifierId.Opponents, "💀");

const Crop = (props: Props) => {
  const context = useGameContext();
  return (
    <button
      disabled={!props.canInteract}
      className="disabled:cursor-not-allowed"
      onClick={() => {
        if (context.selectedCard) {
          context.onSelectSlot(
            context.selectedCard,
            props.boardSlot,
            props.isYourBoard
          );
        }
      }}
    >
      <div
        className={`${
          props.canInteract && "border-2 border-mexicanBone"
        } w-16 h-16 mx-auto bg-yellow-900 hover:bg-yellow-700 flex justify-center items-center rounded-md`}
      >
        <p>{pluck(props.boardSlot.cards, "icon").toString()}</p>
        <p>
          {without(
            map(
              flatten(pluck(props.boardSlot.cards, "modifier")),
              (modifier) => {
                return iconDictionary.get(modifier);
              }
            ),
            undefined
          ).toString()}
        </p>
      </div>
    </button>
  );
};

export default Crop;
