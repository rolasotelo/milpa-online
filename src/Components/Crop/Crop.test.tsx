import React from "react";
import renderer from "react-test-renderer";

import Crop from "./Crop";

describe("<Crop />", () => {
  const defaultProps = {
    text: "ðŸŒ½",
    canInteract: true,
  };
  const wrapper = renderer.create(<Crop {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
