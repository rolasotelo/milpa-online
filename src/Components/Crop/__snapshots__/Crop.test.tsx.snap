// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Crop /> render 1`] = `
<div
  className="w-16 h-16 bg-yellow-900"
>
  ðŸŒ½
</div>
`;
